### ISSUE TITLE
react has-hover of TouchArea for both children and parent element

### ISSUE THREAD

**Issue by wuanzhuan** [slint-ui/slint#4375]
how can i set the row_element's has-hover by TableViewCell's has-hover. Only the TableViewCell has the TouchArea. I need handle hover event for both some TableViewCells and one TableViewRow. the TableViewCell is a child element of TableViewRow.
i try the <=> binding ,it can't work.

    i-scroll-view := ListView {
        for row[idx] in root.rows : row_element := TableViewRow {
            selected: idx == root.current-row;
            even: mod(idx, 2) == 0;
            has-hover: ;

            // order number
            if has-order-number : TableViewCell {
                width: root.order-column-width;
                text: idx;
                index-row: idx;
                index-column: 0;
                horizontal-alignment: center;

                pointer-event(pe, pos) => {
                    root.row-pointer-event(idx, pe, {
                        x: pos.x - root.absolute-position.x,
                        y: pos.y - root.absolute-position.y,
                    });
                }

                clicked => {
                    root.focus();
                    root.set-current-row(idx);
                }

            }
            for cell[index] in row : TableViewCell {
                horizontal-stretch: root.columns[index].horizontal-stretch;
                width: root.columns[index].width;
                text: cell.text;
                index-row: idx;
                index-column: index + 1;
                horizontal-alignment: left;

                pointer-event(pe, pos) => {
                    root.row-pointer-event(idx, pe, {
                        x: pos.x - root.absolute-position.x,
                        y: pos.y - root.absolute-position.y,
                    });
                }

                clicked => {
                    root.focus();
                    root.set-current-row(idx);
                }
            }
        }
    }

**Comment by ogoffart** [comment]
I think you will have to do something like `the_parent.has-hover || the_child.has-hover`

Currently, the mouse area only is consired to have the hover if it is at the top. But it is true that it might make sense to have has-hover to be true also for the ones that are under.  @tronical : what do you think?

**Comment by tronical** [comment]
I agree. If the mouse is over a child of a `TouchArea`, `has-hover` should remain true.

**Comment by 0xrariux** [comment]
I checked this against current master (`bb937076d`) before preparing a fix. If the row's `TouchArea` actually contains the interactive cell's `TouchArea`, both `has-hover` properties remain true over the cell, and the click goes only to the cell. The existing `tests/cases/issues/issue_2717_has-hover.slint` covers this parent/child behavior.

A different structure does reproduce a gap: when the row's `TouchArea` is a background sibling underneath an interactive cell, the cell receives hover and the row's area does not. That is also consistent with the existing test's expectation that only the topmost of overlapping sibling areas is hovered.

Is this issue requesting hover for overlapping sibling `TouchArea`s as well, or specifically for a true ancestor? The sibling interpretation would change hover semantics for unrelated overlapping controls. I have a small regression fixture for either interpretation and can prepare a targeted PR once the intended contract is clear.

**Comment by ogoffart** [comment]
IMHO only ansestors.

Sibling TouchArea are used by pages or some sub-window to block events.
